lib/llvmopencl/TargetAddressSpaces.cc

```diff
@@ -92,6 +92,9 @@ TargetAddressSpaces::runOnModule(llvm::Module &M) {
 
   llvm::StringRef arch(M.getTargetTriple());
 
+  if (arch.startswith("spir"))
+    arch = OCL_KERNEL_TARGET;
```

Comment:
I don't have a suitable system to double check with but I believe this would work with foreign target devices too. However, it does exclude heterogeneous systems.

> So a better fix would be to dig out the device we are compiling to somehow.

I'm thinking that the code that decided the target device ought to be switching the spir triple for the target triple and passing that to the code generator, etc. Can you tell me where that decision is made?